pkg/compactor/planned_jobs_http.go

```
@@ -0,0 +1,178 @@
+	util.RenderHTTPResponse(w, tenantsPageContents{
+		Now:     time.Now(),
+		Tenants: tenants,
+	}, tenantsTemplate, req)
+}
+
+//go:embed planned_jobs.gohtml
+var plannerJobsHTML string
+var plannerJobsTemplateFuncs = template.FuncMap{"add": func(x, y int) int { return x + y }}
+var plannerJobsTemplate = template.Must(template.New("webpage").Funcs(plannerJobsTemplateFuncs).Parse(plannerJobsHTML))
+
+type plannerJobsContent struct {
+	Now                string `json:"now"`
+	BucketIndexUpdated string `json:"bucket_index_updated"`
+
+	Tenant      string                 `json:"tenant"`
+	PlannedJobs []plannedCompactionJob `json:"jobs"`
+
+	ShowBlocks     bool `json:"-"`
+	ShowCompactors bool `json:"-"`
+
+	SplitJobsCount int `json:"split_jobs_count"`
+	MergeJobsCount int `json:"merge_jobs_count"`
+
+	TenantSplitGroups int `json:"tenant_split_groups"`
+	TenantMergeShards int `json:"tenant_merge_shards"`
+	SplitGroups       int `json:"-"`
+	MergeShards       int `json:"-"`
+}
+
+type plannedCompactionJob struct {
+	Key       string      `json:"key"`
+	MinTime   string      `json:"min_time"`
+	MaxTime   string      `json:"max_time"`
+	Blocks    []ulid.ULID `json:"blocks"`
+	Compactor string      `json:"compactor,omitempty"`
+}
+
+func (c *MultitenantCompactor) PlannedJobsHandler(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	tenantID := vars["tenant"]
+	if tenantID == "" {
+		util.WriteTextResponse(w, "Tenant ID can't be empty")
+		return
+	}
+
+	if err := req.ParseForm(); err != nil {
+		util.WriteTextResponse(w, fmt.Sprintf("Can't parse form: %s", err))
+		return
+	}
+
+	showBlocks := req.Form.Get("show_blocks") == "on"
+	showCompactors := req.Form.Get("show_compactors") == "on"
+	tenantSplitGroups := c.cfgProvider.CompactorSplitGroups(tenantID)
+
+	tenantMergeShards := c.cfgProvider.CompactorSplitAndMergeShards(tenantID)
+
+	mergeShards := tenantMergeShards
+	if sc := req.Form.Get("merge_shards"); sc != "" {
+		mergeShards, _ = strconv.Atoi(sc)
```

Shouldn't the potential error be handled here? Just looks like potential error swallowing.

---

> Just looks like potential error swallowing.

That's exactly what it is :)